tscompare/methods.py

```diff
@@ -430,9 +430,12 @@ def f(t):
     # If we did not require this, we could identify swapped samples,
     # but this is out of scope (people could detect this using
     # the shared spans matrix directly).
-    is_sample = np.full(ts.num_samples, False)
+    max_samples = max(ts.num_samples, other.num_samples)
+    is_sample = np.full(max(ts.num_nodes, other.num_nodes), False)
     is_sample[samples] = True
-    shared_spans.data[~np.logical_xor(is_sample[row_ind], is_sample[col_ind])] = 0.0
+    index_contains_sample = is_sample[row_ind]
+    index_not_equal = ~np.equal(row_ind, col_ind)
+    shared_spans.data[np.logical_and(is_sample[row_ind], index_not_equal)] = 0.0
```

Comment:
Ah, whoops - good catch. But I think it's not quite there: right now there's no need to have `is_sample` the length of `other.num_nodes`, because it's only ever indexed by `row_ind`. Actually, this will produce an error if `ts.num_nodes != other.num_nodes` - both on indexing by `row_ind` and also since `np.equal( )` requires equal lengths.

(Also,  `index_contains_sample` isn't used?)

Reply:
So - looks like we need a test where the number of nodes in the two tree sequences isn't equal?

Reply:
Hah, as you said, I'm totally wrong here. Never mind. =)